### PR TITLE
chore: [Destinations] Improve Test for Destination Service Option Combinations

### PR DESCRIPTION
**Pull request by MatKuhr**
## Context

Follow up to #190 

This PR hardens the `GetOrComputeSingleDestinationCommandWithoutAllDestinationsTest` by mocking the `DestinationServiceV1Reponse` (instead of `HttpDestination`). That way the relevant logic from `DestinationServiceFactory` is now tested (instead of mocked).

### CODE REVIEW

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultHttpDestinationBuilderProxyHandler.java [hunk 1]
```diff
@@ -71,16 +71,14 @@ DefaultHttpDestination handle( @Nonnull final DefaultHttpDestination.Builder bui
...
-    // *                    + SUB_ONLY        + NO_TENANT + *        -> (should never happen as it should fail on destination loading)
-    // *                    + SUB_ONLY        + TENANT_PR + *        -> (should never happen as it should fail on destination loading)
-    // BasicAuth            + ALWAYS_PROVIDER + *         + NO_USER  -> TECHNICAL_USER_PROVIDER
-    // BasicAuth            + CURRENT_TENANT  + *         + NO_USER  -> TECHNICAL_USER_CURRENT_TENANT
-    // BasicAuth            + SUB_ONLY        + TENANT_T1 + NO_USER  -> TECHNICAL_USER_CURRENT_TENANT
-    // PrincipalPropagation + *               + *         + NO_USER  -> (fail, on getHeaders())
-    // PrincipalPropagation + ALWAYS_PROVIDER + TENANT_PR + USER_PR  -> TECHNICAL_USER_CURRENT_TENANT (compatibility strategy) or NAME_USER_CURRENT_TENANT (recommended strategy)
-    // PrincipalPropagation + ALWAYS_PROVIDER + TENANT_T1 + *        -> (fail, on getHeaders())
-    // PrincipalPropagation + CURRENT_TENANT  + *         + *        -> TECHNICAL_USER_CURRENT_TENANT
+    // Auth type            | destination tenant | PP mode           | result
```
Comment (MatKuhr) [inline cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultHttpDestinationBuilderProxyHandler.java:74]: driveby, changed to reflect actual behavior of the method

File: cloudplatform/connectivity-destination-service/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/GetOrComputeSingleDestinationCommandWithoutAllDestinationsTest.java [hunk 2]
```diff
@@ -267,9 +265,14 @@ private void runTest( TestCase testCase )
         } else if( maybeDestination.isSuccess() && !testCase.isCommandExecutionExpectedToSucceed() ) {
             softly.fail("Expected command execution to fail, but it succeeded.");
         }
-        softly.assertThat(maybeDestination.get()).isEqualTo(testCase.getDestination());
+        // softly.assertThat(maybeDestination.get()).isEqualTo(testCase.getExpectedDestination());
```
Comment (newtork) [inline cloudplatform/connectivity-destination-service/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/GetOrComputeSingleDestinationCommandWithoutAllDestinationsTest.java:268]: (Major)

Please uncomment, fix or remove this line

File: cloudplatform/connectivity-destination-service/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/GetOrComputeSingleDestinationCommandWithoutAllDestinationsTest.java [hunk 3]
```diff
@@ -204,15 +201,16 @@ static Collection<TestCase> createTestParameters()
...
+    private DestinationService destinationService;
     private SoftAssertions softly;
 
     @BeforeEach
     void setup()
     {
         destinationCache = Caffeine.newBuilder().build();
         isolationLocks = Caffeine.newBuilder().build();
-        destinationRetriever = (BiFunction<String, DestinationOptions, Destination>) mock(BiFunction.class);
+        final DestinationServiceAdapter adapter = mock(DestinationServiceAdapter.class);
```
Comment (newtork) [inline cloudplatform/connectivity-destination-service/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/GetOrComputeSingleDestinationCommandWithoutAllDestinationsTest.java:212]: (Major)

Unused variable.

File: cloudplatform/connectivity-destination-service/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/GetOrComputeSingleDestinationCommandWithoutAllDestinationsTest.java [hunk 4]
```diff
@@ -492,56 +480,88 @@ TestCase forAuthenticationTypeAndOptions(
...
-                    principal,
-                    isCommandCreationExpectedToSucceed,
-                    isCommandExecutionExpectedToSucceed,
                     expectedIsolationKey,
                     expectedDestinationKey);
             }
 
-            private static Destination prepareDestination( AuthenticationType authType, Principal principal )
+            private TestCase finaliseTestCase(
+                String destinationName,
```
Comment (newtork) [inline cloudplatform/connectivity-destination-service/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/GetOrComputeSingleDestinationCommandWithoutAllDestinationsTest.java:512]: (Minor)

We're not doing `final` anymore? :)

File: cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultHttpDestinationBuilderProxyHandler.java [hunk 5]
```diff
@@ -71,16 +71,14 @@ DefaultHttpDestination handle( @Nonnull final DefaultHttpDestination.Builder bui
...
+    // *                    + NO_TENANT          + *                -> TECHNICAL_USER_PROVIDER
+    // *                    + TENANT_T1          + *                -> TECHNICAL_USER_CURRENT_TENANT
+    // *                    + TENANT_T1          + *                -> TECHNICAL_USER_CURRENT_TENANT
+    // PrincipalPropagation + NO_TENANT          + TOKEN_FORWARDING -> TECHNICAL_USER_PROVIDER
+    // PrincipalPropagation + NO_TENANT          + TOKEN_EXCHANGE   -> NAMED_USER_CURRENT_TENANT
```
[the remark this answers is not in the document]
Comment (MatKuhr) [reply cloudplatform/cloudplatform-connectivity/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DefaultHttpDestinationBuilderProxyHandler.java]: it means provider tenant, which is represented by an empty string, I updated the table to reflect this better

**Review by newtork — APPROVED** [review]
Not  a huge fan of all these conditions in a unit test. That can make it difficult to understand the exact assertions for the test cases from just reading the code.

Otherwise LGTM.
